frontend/src/components/transactions/ActionRow.tsx

```diff
@@ -33,17 +31,8 @@ export default class extends React.Component<Props, State> {
       action,
       showDetails
     } = this.props;
-
-    let actionKind: keyof T.Action;
-    let actionArgs: T.Action;
-    if (typeof action === "string") {
-      actionKind = action;
-      actionArgs = {} as T.Action;
-    } else {
-      actionKind = Object.keys(action)[0] as keyof T.Action;
-      actionArgs = action[actionKind] as T.Action;
-    }
-
+    let actionKind = action.kind as keyof T.Action;
```

Comment:
Once we address the comments in `explorer-wamp/transactions.ts`, we should not have a need for this type casting.